### ISSUE TITLE
Feature Request: Make build support files optional

### ISSUE THREAD

**Issue by wienczny** [swagger-api/swagger-codegen#3724]
I'm using the `swagger-codegen-maven-plugin` to generate clients for my swagger apis. As the swagger description may change in the future the code generator is run for every build. The generated code is directly referenced by the same maven project that runs the code generator. Because of that the generated code contains a maven, gradle and sbt project and more files that are not required. The only files required by my project are those generated into src/main/java. Those are added to the maven build path and added to the jar.

Please allow to disable the code generation for optional files.

**Comment by wing328** [comment]
Please update .swagger-codege-ignore to skip the files that you don't want to generate.

**Comment by wienczny** [comment]
Using .swagger-codegen-ignore is not an ideal solution for this issue but a nice workaround.

I think adding a switch to the codegen is still required:

- You need additional maven plugins to copy .swagger-codegen-ignore to the output directory
- Not all ignored files and directories are skipped by the codegen. The gradle directory is still generated although /gradle is in the ignore file.

**Comment by brentryan** [comment]
+1

**Comment by wing328** [comment]
@wienczny agreed with you that my suggestion is more of a workaround. Would you have time to implement the option?

> You need additional maven plugins to copy .swagger-codegen-ignore to the output directory

You can customize the swagger-codegen-ignore template to begin with and pass the customized templates via the `-t` option.

**Comment by wienczny** [comment]
I would suggest this:

- Add an enumeration (SourceType) for types of generated files with
 - Source
 - Test
 - Project
 - ...
- Add a field with type SourceType to SupportingFile.
- Set<SourceType> to CodegenConfigurator
- Filter SupportingFile in DefaultGenerator.java:568.

I found a duplicate at #3737
